### DISCUSSION TITLE
Camera suddenly changes it's position and rotation after some time in WebXR but not in model-viewer's WebXR

### DISCUSSION THREAD

**Discussion by SuperCaptainWOW1** [google/model-viewer#4484] [Q&A] [answered]
I am having a problem with WebXR. When I launch a WebXR AR session, my objects are showing in one certain place. But after some time, apparently, when room scan is fully complete, they suddenly teleport to my right side and appear to have different rotations. I checked, the position and rotation of the objects themselves stays intact. But position and rotation of the camera alters significantly.

You can see on this screenshot the rotation of the camera being printed every frame.
![image](https://github.com/google/model-viewer/assets/44094081/047c35fd-e3ba-4a26-90d6-595bacdc4eb4)

This scenario reproduces in every official WebXR example, my app and even a clean three js project with zero unnecessary code. The only explanation I found was basically - WebXR isn't perfect. But this problem is completely missing in model viewer. 

So I'm wondering if this is something that was purposely handled by some specific workarounds? And if not what could possibly solve this issue?

Was tested on Google Pixel 7 and Samsung Galaxy Z Fold3, in Chrome and Samsung Internet browsers.

Thanks in advance.

**Comment by elalish** [comment]
Huh, it's been awhile since I wrote that code, but I'm glad it's working at least! Maybe you'll find your answer in [here](https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/three-components/ARRenderer.ts)?

**Reply by SuperCaptainWOW1** [reply, marked as answer]
Thanks for the hint! I actually found what resolves the issue in this file. But it was not what I expected.
So if anyone having the same problem. Solution for me was to just add _'light-estimation'_ to the list of required features while starting the new WebXR session:
`
navigator.xr.requestSession("immersive-ar", {
  requiredFeatures: ['light-estimation']
});
`
It seems that without this feature, scanning isn't good enough to produce a valid result right away, so the camera repositions itself after some time to match the new belated data.

**Reply by elalish** [reply]
@bialpio Is this expected behavior or an indication of a WebXR/ARCore bug?

**Comment by bigbea1r** [comment]
SuperCaptainWOW, please feel free to contact me. I need help. Goal: hack modelviewer. access the modelviewer through three.js to create an editor based on the modelviewer. Three.js is an important part of the project.

Пожалуйста, можете связаться со мной. Мне нужна помощь. Цель: взломать modelviewer. обращаться к modelviewer через three.js, чтобы создать редактор на основе modelviewer. Three.js важная часть проекта. 

 telegram: https://t.me/f_game_over
